src/extension/monitor/core/src/main/java/org/geoserver/monitor/ows/MonitorCallback.java

```diff
@@ -106,19 +105,36 @@ public Object operationExecuted(Request request, Operation operation, Object res
     }
 
     public void finished(Request request) {
-        if (request.getError() != null) {
-            RequestData data = monitor.current();
-            if (data == null) {
-                // will happen in cases where the filter is not active
-                return;
-            }
 
+        RenderTimeStatistics statistics =
+                (RenderTimeStatistics)
+                        request.getHttpRequest().getAttribute(RenderTimeStatistics.ID);
+        RequestData data = monitor.current();
+        if (data == null) {
+            // will happen in cases where the filter is not active
+            return;
+        }
+        if (request.getError() != null) {
             data.setStatus(Status.FAILED);
             data.setErrorMessage(request.getError().getLocalizedMessage());
             data.setError(request.getError());
-
             monitor.update();
         }
+
+        if (statistics != null) {
+            List<Long> renderingTimeLayers = new ArrayList<>(statistics.getRenderingLayersIdxs().size());
+            data.setLabellingProcessingTime(statistics.getLabellingTime());
+            data.setResources(statistics.getLayerNames());
```

Comment:
Why the update? Is there a risk of the resource list already been collected not matching the list in the stats? That is worrysome.

Reply:
fixed